From 8f699f70262d6acb919fc3db2e83930d9f271db4 Mon Sep 17 00:00:00 2001
From: "Darryl L. Pierce" <dpierce@redhat.com>
Date: Tue, 1 Apr 2014 16:12:48 -0400
Subject: [PATCH 5/7] QPID-5656: Updated Ruby bindings to build under CMake 3.0

CMake 3.0 has reverted to the CMake 2.6 naming for libraries. prepending
"lib" to only the Ruby library names.
---
 qpid/cpp/bindings/qmf2/ruby/CMakeLists.txt | 6 ++++--
 qpid/cpp/bindings/qpid/ruby/CMakeLists.txt | 6 ++++--
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/qpid/cpp/bindings/qmf2/ruby/CMakeLists.txt b/qpid/cpp/bindings/qmf2/ruby/CMakeLists.txt
index 4ea61ab..19fe4bf 100644
--- a/qpid/cpp/bindings/qmf2/ruby/CMakeLists.txt
+++ b/qpid/cpp/bindings/qmf2/ruby/CMakeLists.txt
@@ -36,7 +36,8 @@ swig_link_libraries(cqmf2_ruby qmf2 ${RUBY_LIBRARY})
 ##----------------------------------
 ## Install the complete Ruby binding
 ##----------------------------------
-if ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8))
+if ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8) OR
+    (${CMAKE_MAJOR_VERSION} EQUAL 3))
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cqmf2_ruby.so
         RENAME cqmf2.so
         DESTINATION ${RUBY_PFX_ARCH_DIR}
@@ -48,5 +49,6 @@ else()
         DESTINATION ${RUBY_PFX_ARCH_DIR}
         COMPONENT ${QPID_COMPONENT_CLIENT}
         )
-endif ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8))
+endif ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8) OR
+       (${CMAKE_MAJOR_VERSION} EQUAL 2))
 
diff --git a/qpid/cpp/bindings/qpid/ruby/CMakeLists.txt b/qpid/cpp/bindings/qpid/ruby/CMakeLists.txt
index 60c6044..f40af7a 100644
--- a/qpid/cpp/bindings/qpid/ruby/CMakeLists.txt
+++ b/qpid/cpp/bindings/qpid/ruby/CMakeLists.txt
@@ -51,7 +51,8 @@ set_source_files_properties(${swig_generated_file_fullname} PROPERTIES COMPILE_F
 ##----------------------------------
 ## Install the complete Ruby binding
 ##----------------------------------
-if ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8))
+if ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8) OR
+    (${CMAKE_MAJOR_VERSION} EQUAL 3))
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cqpid_ruby.so
         RENAME cqpid.so
         DESTINATION ${RUBY_PFX_ARCH_DIR}
@@ -63,7 +64,8 @@ else()
         DESTINATION ${RUBY_PFX_ARCH_DIR}
         COMPONENT ${QPID_COMPONENT_CLIENT}
         )
-endif ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8))
+endif ((${CMAKE_MAJOR_VERSION} EQUAL 2) AND (${CMAKE_MINOR_VERSION} LESS 8) OR 
+       (${CMAKE_MAJOR_VERSION} EQUAL 3))
 
 add_custom_command(OUTPUT ${GEM_BINDINGS_SOURCE_FILE}
                    COMMAND cp ${swig_generated_file_fullname} ${GEM_BINDINGS_SOURCE_FILE}
-- 
1.9.0

